python/pyarrow/_dataset.pyx

```diff
@@ -909,13 +909,24 @@ cdef class ParquetFileFragment(FileFragment):
 
     def __reduce__(self):
         buffer = self.buffer
+        if self.row_groups is not None:
```

Comment:
When I started working on the python binding, the compiler complained that RowGroupInfo is not pickleable. I'm not sure why this didn't fail before; we've never had RowGroupInfo.__reduce__ and ParquetFileFormat.make_fragment only accepts a list of integers for the row_groups argument

Reply:
Yeah, I actually realized we were not yet pickling the row group id's when discussing this in the meeting we had, and was planning to open a JIRA / do a quick PR, but you already fixed it ;)

(it didn't fail because we simply didn't include any row group information in the serialization)

Only preserving the rowgroup id's (as you do here) should be sufficient for now.

Reply:
Note that (due to stats containing scalars) pickling of RowGroupInfo depends on pickling of Scalars. Added ARROW-9394

Reply:
Indeed. Based on my discussions with @rjzamora earlier today, the row group id's are the most critical to serialize (which this PR thus already fixes), and should cover the short-term needs for dask's use cases.